docker/Dockerfile

```
@@ -1,19 +1,19 @@
 FROM alpine:edge AS builder
-RUN apk add --no-cache crystal shards libc-dev \
-    yaml-dev libxml2-dev sqlite-dev zlib-dev openssl-dev \
-    sqlite-static zlib-static openssl-libs-static
 WORKDIR /invidious
 COPY ./shard.yml ./shard.yml
-RUN shards update && shards install
-RUN apk add --no-cache curl && \
+RUN apk add crystal shards libc-dev \
+    yaml-dev libxml2-dev sqlite-dev zlib-dev openssl-dev \
+    yaml-static sqlite-static zlib-static openssl-libs-static && \
```

It seems that `yaml-dev` and `sqlite-dev` are not necessary (from my testing with `docker-compose`).
```suggestion
RUN apk add crystal shards libc-dev \
    libxml2-dev zlib-dev openssl-dev \
    yaml-static sqlite-static zlib-static openssl-libs-static && \
```

---

strange, I would think linking would need header files.

---

I think it's because the YAML and SQLite libraries are only statically linked and so the dynamic libraries in `yaml-dev` and `sqlite-dev` are not used.

---

I mean yaml.h etc. or am I misunderstanding how linking works?

---

> I mean yaml.h etc. or am I misunderstanding how linking works?

If I understand correctly `yaml.h` is the source code header file which can be compiled to a `.so` for dynamic linking or a `.a` for static linking.